src/csprojWriter.ts

```diff
@@ -0,0 +1,227 @@
+
+            if (itemType === BuildActions.Compile && includePath.endsWith('.xaml.cs')) {
+                const pagePath = includePath.replace('.cs', '');
+                const pageBuildAction = await this.get(projPath, pagePath);
+
+                if (pageBuildAction === BuildActions.Page) item[itemType].DependentUpon = path.basename(pagePath);
+            } else if (itemType === BuildActions.Page) {
+                item[itemType].SubType = 'Designer';
+                item[itemType].Generator = 'MSBuild:Compile';
+            }
+
+            if (itemGroup) {
+                const actions: Array<any> | undefined = itemGroup[itemType];
+
+                if (actions) {
+                    actions.push(item[itemType]);
+                } else {
+                    const array: Array<any> = [];
+
+                    array[0] = item[itemType];
+
+                    itemGroup[itemType] = array;
+                }
+            } else {
+                const array: Array<any> = [];
+
+                array[0] = item[itemType];
+
+                items.push({ [itemType]: array });
+            }
+        }
+
+        await fs.writeFile(projPath, new Builder().buildObject(parsedXml));
+    }
+
+    public async get(projPath: string, itemPath: string): Promise<BuildActions | undefined> {
+        itemPath = this.fixItemPath(projPath, itemPath);
+
+        const parsedXml = await this.parseProjFile(projPath);
+
+        if (!parsedXml) return;
+
+        const items: Array<any> = parsedXml.Project.ItemGroup;
+
+        if (items) {
+            for (const item of items) {
+                const actions = Object.values(item)[0] as Array<any>;
+
+                if (actions) {
+                    for (const action of actions) {
+                        if (action.$.Include  === itemPath) {
+                            return BuildActions[Object.getOwnPropertyNames(item)[0] as keyof typeof BuildActions];
+                        }
+                    }
+                }
+            }
+        }
+
+        return;
+    }
+
+    public async remove(projPath: string, itemPath: string): Promise<void> {
+        let isDir = false;
+
+        try {
+            const fileStat = await fs.lstat(itemPath);
+
+            isDir = fileStat.isDirectory();
+        } catch { }
+
+        itemPath = this.fixItemPath(projPath, itemPath);
+
+        const parsedXml = await this.parseProjFile(projPath);
+
+        if (!parsedXml) return;
+
+        const items: Array<any> = parsedXml.Project.ItemGroup;
+
+        for (const item of items) {
+            const actionsArray = Object.values(item) as Array<Array<any>>;
+
+            for (const actions of actionsArray) {
+                for (const action of actions) {
+                    const include = action.$.Include;
+
+                    if (include === itemPath || (isDir && include.startsWith(itemPath))) {
+                        actions.splice(action, 1);
```

Comment:
The object `action` can't be used as an index. Use `actions.splice(actions.indexOf(action), 1);` instead.